AdminRouter/start-windows-build.ps1

```diff
@@ -0,0 +1,266 @@
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$LocalFilesPath,
+        [Parameter(Mandatory=$true)]
+        [string]$RemoteFilesPath
+    )
+    Write-Output "Started copying files from $LocalFilesPath to remote location at ${server}:${RemoteFilesPath}"
+    Start-SCPCommand -Server $REMOTE_LOG_SERVER -User $REMOTE_USER -Key $env:SSH_KEY `
+                     -LocalPath $LocalFilesPath -RemotePath $RemoteFilesPath
+}
+
+function New-RemoteDirectory {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$RemoteDirectoryPath
+    )
+    $remoteCMD = "if [[ -d $RemoteDirectoryPath ]]; then rm -rf $RemoteDirectoryPath; fi; mkdir -p $RemoteDirectoryPath"
+    Start-SSHCommand -Server $REMOTE_LOG_SERVER -User $REMOTE_USER -Key $env:SSH_KEY -Command $remoteCMD
+}
+
+function New-RemoteSymlink {
+    Param(
+        [Parameter(Mandatory=$true)]
+        [string]$RemotePath,
+        [Parameter(Mandatory=$false)]
+        [string]$RemoteSymlinkPath
+    )
+    $remoteCMD = "if [[ -h $RemoteSymlinkPath ]]; then unlink $RemoteSymlinkPath; fi; ln -s $RemotePath $RemoteSymlinkPath"
+    Start-SSHCommand -Server $REMOTE_LOG_SERVER -User $REMOTE_USER -Key $env:SSH_KEY -Command $remoteCMD
+}
+
+function Get-AdminRouterBuildRelativePath {
+    $repositoryOwner = $GitURL.Split("/")[-2]
+    $commitID = Get-LatestCommitID
+    return "$repositoryOwner/$Branch/$commitID"
+}
+
+function Get-RemoteBuildDirectoryPath {
+    $relativePath = Get-AdminRouterBuildRelativePath
+    return "$REMOTE_ADMINROUTER_BUILD_DIR/$relativePath"
+}
+
+function Get-BuildOutputsUrl {
+    $relativePath = Get-AdminRouterBuildRelativePath
+    return "$ADMINROUTER_BUILD_BASE_URL/$relativePath"
+}
+
+function New-RemoteLatestSymlinks {
+    $remoteDirPath = Get-RemoteBuildDirectoryPath
+    $baseDir = (Split-Path -Path $remoteDirPath -Parent) -replace '\\', '/'
+    New-RemoteSymlink -RemotePath $remoteDirPath -RemoteSymlinkPath "$baseDir/latest"
+    $repoDir = (Split-Path -Path $baseDir -Parent) -replace '\\', '/'
+    New-RemoteSymlink -RemotePath $remoteDirPath -RemoteSymlinkPath "$repoDir/latest"
+}
+
+function Start-LogServerFilesUpload {
+    $consoleUrl = "${JENKINS_SERVER_URL}/job/${env:JOB_NAME}/${env:BUILD_NUMBER}/consoleText"
+    Start-FileDownload -Force -URL $consoleUrl -Destination "$ADMINROUTER_BUILD_LOGS_DIR\console-jenkins.log"
+    $remoteDirPath = Get-RemoteBuildDirectoryPath
+    New-RemoteDirectory -RemoteDirectoryPath $remoteDirPath
+    Copy-FilesToRemoteServer "$ADMINROUTER_BUILD_OUT_DIR\*" $remoteDirPath
+    $buildOutputsUrl = Get-BuildOutputsUrl
+    $global:PARAMETERS["BUILD_OUTPUTS_URL"] = $buildOutputsUrl
+    Write-Output "Build artifacts can be found at: $buildOutputsUrl"
+    if($global:PARAMETERS["BUILD_STATUS"] -eq "PASS") {
+        New-RemoteLatestSymlinks
+    }
+ }
+
+function Start-EnvironmentCleanup {
+    # Stop any potential hanging process
+    $processes = @('go', 'bash')
```

Comment:
Does this spawn any of the processes: `go` or `bash` ?

Reply:
Not at all. I will remove those copied over lines.
Thanks for catching this